python_modules/dagster/dagster/components/resolved/core_models.py

```diff
@@ -56,10 +55,11 @@ def resolve_backfill_policy(
     raise ValueError(f"Invalid backfill policy: {backfill_policy}")
 
 
-@dataclass
-class OpSpec(Resolvable):
+class OpSpec(Model, Resolvable):
```

Comment:
We probably need a way to ensure these params are all consumed in each callsite? e.g. right now if the user would set `description` here it wouldn't flow through in https://github.com/dagster-io/dagster/blob/master/python_modules/libraries/dagster-dbt/dagster_dbt/components/dbt_project/component.py#L183

I might give that a stab in a stacked pr

Reply:
I actually want to make it a norm to stack on `FunctionComponent` more so that type of thing is never miessed. That will be harder with dbt but for example I believe that the sling component can be stacked on FunctionComponent fairly easily.